activerecord/lib/active_record/connection_adapters/abstract/schema_definitions.rb

```diff
@@ -171,6 +171,20 @@ def initialize(
         end
       end
 
+      def add(table_name, connection)
```

Comment:
Can't we just make `add_to` behaves like this one? Is there any case where we want the table object and ignore those options we are trying to not ignore here?

Reply:
Good question; the reason that I chose to add a new method is so adding references is consistent with every other table modification: that they have separate code paths for when they are called inside or outside a `change_table` block, which really matters because of the "batch" option.

Context: In writing https://github.com/rails/rails/pull/43576, I realized that inside a `change_table` block called with `batch: true`, migration methods are called on a migration recorder, rather than on a table (so they can be combined later into a single update query). I wasn't even sure it was possible to have batch update mode correctly respond to `if_exists`/`if_not_exists` without being error prone, which is why I chose to raise an exception when those options are passed to a method that will ignore them.

`add_reference` was just coded in such a way that it always called its methods directly on a `Table` both inside and outside a `change_table` block (unlike all the other table modification methods). Since I already didn't think it was possible to make changes to batch update mode, I figured that adding a separate method for use outside of a `change_table` was the next best option.

Reply:
This makes sense to me, and we also need this to unblock our weekly Rails bump. @rafaelfranca if you have additional changes let me know and I'll make them.